tests/maintel/tma/test_random_walk.py

```diff
@@ -121,13 +121,13 @@ def actualPosition(self):
             )
 
             sky_offsets = []
-            async for i, az, el, off in self.script.random_walk_azel_by_time():
-                sky_offsets.append(off)
+            async for data in self.script.random_walk_azel_by_time():
+                sky_offsets.append(data.offset)
 
                 # TODO: This should not be necessary
                 # However, if I remove it the loop keeps running forever
                 # It does not seem to happen when running the loop w/ hardware
-                if i == 50:
+                if data.counter == 50:
```

Comment:
Regarding your TODO: I would fully expect this to run forever unless `random_walk_azel_by_time` has an intrinsic limit in the # of targets it will return. 

If it does, then this tests shows that limit is broken. if it does not, then you need to decide how many targets you want.

Please remove the TODO comment -- after fixing the method, if necessary.

Reply:
The unit test is running fine now.